### PR TITLE
KSTREAMS-2871, KSTREAMS-2870, KSTREAMS-2868: Fix test failures

### PR DESCRIPTION
**Pull request by cadonna**
- Changed handling of result of `get` on user-defined store since
   stores return a value and a timestamp since 2.3
- Set STATE_DIR_CONFIG to temp directory.

### CODE REVIEW

**Comment by guozhangwang** [comment]
The change lgtm.

Is this failure relevant though?

```
io.confluent.examples.streams.ValidateStateWithInteractiveQueriesLambdaIntegrationTest.shouldComputeMaxValuePerKey
```

**Comment by cadonna** [comment]
> The change lgtm.
> 
> Is this failure relevant though?
> 
> ```
> io.confluent.examples.streams.ValidateStateWithInteractiveQueriesLambdaIntegrationTest.shouldComputeMaxValuePerKey
> ```

No, `io.confluent.examples.streams.ValidateStateWithInteractiveQueriesLambdaIntegrationTest.shouldComputeMaxValuePerKey` fails because of the bug, that is fixed in https://github.com/apache/kafka/pull/6808.

File: src/test/java/io/confluent/examples/streams/CustomStreamTableJoinIntegrationTest.java [hunk 1]
```diff
@@ -319,14 +320,14 @@ public void shouldTriggerStreamTableJoinFromTable() throws Exception {
...
-        private KeyValueStore<String, Long> tableStore;
+        private KeyValueStore<String, ValueAndTimestamp<Long>> tableStore;
         private ProcessorContext context;
 
         @SuppressWarnings("unchecked")
         @Override
         public void init(final ProcessorContext context) {
           streamBufferStore = (KeyValueStore<String, Pair<Double, Instant>>) context.getStateStore(streamBufferStoreName);
-          tableStore = (KeyValueStore<String, Long>) context.getStateStore(tableStoreName);
+          tableStore = (KeyValueStore<String, ValueAndTimestamp<Long>>) context.getStateStore(tableStoreName);
```
Comment (vvcephei) [inline src/test/java/io/confluent/examples/streams/CustomStreamTableJoinIntegrationTest.java]: I'm not too comfortable with this API breakage, but it seems like it's too late to do anything about it now...

Comment (mjsax) [reply src/test/java/io/confluent/examples/streams/CustomStreamTableJoinIntegrationTest.java]: Well. Thinking about it, we might not want to recommend this pattern in the first place... \cc @miguno 

It's actually quite a hack to access to store of a DSL operator in a custom transformer from my point of view... Thoughts?

**Comment by mjsax** [comment]
```
io.confluent.examples.streams.ValidateStateWithInteractiveQueriesLambdaIntegrationTest.shouldComputeMaxValuePerKey

Exception caught in process. taskId=0_0, processor=KSTREAM-SOURCE-0000000000, topic=inputTopic, partition=0, offset=0, stacktrace=org.apache.kafka.streams.errors.StreamsException: A serializer (value: org.apache.kafka.streams.state.internals.ValueAndTimestampSerializer) is not compatible to the actual value type (value type: org.apache.kafka.streams.state.ValueAndTimestamp). Change the default Serdes in StreamConfig or provide correct Serdes via method parameters.
 at org.apache.kafka.streams.state.StateSerdes.rawValue(StateSerdes.java:195)
 at org.apache.kafka.streams.state.internals.MeteredWindowStore.put(MeteredWindowStore.java:138)
 at org.apache.kafka.streams.processor.internals.ProcessorContextImpl$WindowStoreReadWriteDecorator.put(ProcessorContextImpl.java:533)
 at org.apache.kafka.streams.kstream.internals.KStreamWindowAggregate$KStreamWindowAggregateProcessor.process(KStreamWindowAggregate.java:138)
 at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
 at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
 at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
 at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:133)
 at org.apache.kafka.streams.processor.internals.SourceNode.process(SourceNode.java:87)
 at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:366)
 at org.apache.kafka.streams.TopologyTestDriver.pipeInput(TopologyTestDriver.java:419)
 at io.confluent.examples.streams.IntegrationTestUtils.produceKeyValuesSynchronously(IntegrationTestUtils.java:411)
 at io.confluent.examples.streams.ValidateStateWithInteractiveQueriesLambdaIntegrationTest.shouldComputeMaxValuePerKey(ValidateStateWithInteractiveQueriesLambdaIntegrationTest.java:114)
```

@cadonna seems relevant.

**Comment by mjsax** [comment]
Triggered a rebuild -- hope that the fix from apache/kafka#6808 is already synced.

**Pull request merged**